HW2/HW2.sql

```diff
@@ -0,0 +1,77 @@
+USE baseball;
+
+# You have probably already corrected this
+#ALTER TABLE inning MODIFY COLUMN game_id INT UNSIGNED NOT NULL;
+
+CREATE INDEX game_id ON batter_counts(game_id);
+
+#assignment
+#Calculate the batting average using SQL queries for every player
+CREATE TABLE batter_ba as
+SELECT a.batter,
+       CASE WHEN sum(a.atBat)>0 THEN
+       format(sum(a.Hit) / sum(a.atBat),3)
+           ELSE NULL END as Batting_Average
+from batter_counts a
+group by a.batter;
+
+#Annual batting averages and order nicely
+CREATE TABLE batter_annual_ba as
+SELECT a.batter,
+       year(b.local_date)        as YEAR,
+       CASE WHEN sum(a.atBat) > 0 THEN
+       format(sum(a.Hit) / sum(a.atBat),3)
+           ELSE NULL END as Batting_Average
+from batter_counts a,
+     game b
+WHERE a.game_id = b.game_id
+group by a.batter,
+         year
+order by 1, 2;
```

Comment:
using numbers here is a little ambiguous.  Might be easier to read if you use the names of the columns instead or leave a comment.

Reply:
Agreed!